catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/CachedClientPool.java

```diff
@@ -136,6 +147,13 @@ private static ThreadFactory newDaemonThreadFactory() {
 
   public void close() {
     clientPoolCache.invalidateAll();
+
+    // Let the remove listener to close the HiveClientPool first then close the class loader.
+    try {
+      Thread.sleep(500);
+    } catch (Exception e) {
+      // Ignore
+    }
```

Comment:
The code here is not robust enough, you need to have a check mechanism to make sure client pool is really closed beforehand.